### PR TITLE
Simple tests for IdSet

### PR DESCRIPTION
**Pull request by kshyatt**
Appears to be [missing](https://codecov.io/gh/julialang/julia/src/master/base/idset.jl) tests.

### CODE REVIEW

File: test/sets.jl [hunk 1]
```diff
@@ -709,3 +709,20 @@ Base.IteratorSize(::Type{<:OpenInterval}) = Base.SizeUnknown()
...
+    A = filter!(isodd, A)
+    @test A ⊆ B
+    @test !(B ⊆ A)
+    @test !isempty(A)
+    a = pop!(A, 1)
+    @test a == 1
+    @test !isempty(A)
+    A = empty!(A)
+    @test isempty(A)
+end
```
Comment (JeffBezanson) [inline test/sets.jl:728]: Would be good to test that elements are compared with `===`, e.g. using an `IdSet{Vector{Int}}`.

Comment (kshyatt) [reply test/sets.jl:732]: Better now?

**Comment by kshyatt** [comment]
pinging again gently for a second review :)

File: test/sets.jl (hunk 1 above)
Comment (timholy) [reply test/sets.jl:732]: Maybe something like
```julia
julia> s = Base.IdSet{Vector{Int}}([a, b])
Base.IdSet{Array{Int64,1}} with 2 elements:
  [1]
  [2]

julia> a ∈ s
true

julia> c = [1]
1-element Array{Int64,1}:
 1

julia> c ∈ s
false
```

Comment (Keno) [reply test/sets.jl:732]: Would probably be good to also add something that is `== a`, but not `=== a` to test that it's not just a regular `Dict`.

Comment (fingolfin) [reply test/sets.jl:732]: To make such a test, I guess one could e.g. use `Vector{Int}` keys:
```julia
julia> dict = IdDict([0] => 1, [0] => 2)
IdDict{Array{Int64,1},Int64} with 2 entries:
  [0] => 1
  [0] => 2
```

Comment (fingolfin) [reply test/sets.jl:732]: That said, this PR looks like a nice improvement, so even without that, it'd be nice to merge it (and improve it later?) Depending on whether @kshyatt still has any interest on "finishing" this or not, it'd be a shame to let this go to waste?

**Comment by kshyatt** [comment]
I'm still interested :). When I get a free moment I'll update it.

**Comment by StefanKarpinski** [comment]
Anything to be done here? Does this still need review or can we just merge it?

**Comment by kshyatt** [comment]
Updated this PR. I think we can just merge it as is if it passes tests and make further improvements later.

**Comment by kshyatt** [comment]
Got the ok from @StefanKarpinski to merge

**Pull request merged**